test/com/twu/biblioteca/WelcomeTest.java

```diff
@@ -0,0 +1,25 @@
+package com.twu.biblioteca;
+
+
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.Assert.assertEquals;
+
+public class WelcomeTest {
+
+    @Test
+    public void shouldWriteWelcomeMessage() {
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(output);
+        System.setOut(ps);
```

Comment:
Nice!  Normally you wouldn't have such a lopsided test:code ratio (6:1), but it's good to see you found out how to stub the PrintStream for this and I like the usage of the System.getProperty for a platform-independent line separator.

Reply:
Question: is there another way to get control over the PrintStream that doesn't change it for the _entire_ Java process?  PrintStream is effectively an example of the Singleton pattern, so when you change it in this test, you will be changing it for all Java code running inside the same JVM that the test is running in.  This can be a problem if you have a lot of tests running in parallel inside the same JVM instance.  

Is there a more local contained way to assert the output without changing PrintStream directly?  Hint: there is :-)

Reply:
Thanks, most code online was suggesting using \n but it would look different on different platforms so that wasn't optimal.

Been struggling to work out a solution to not changing the PrintStream directly, I'll have to come back it later so I can move onto the next user story.